saleor/graphql/product/mutations.py

```diff
@@ -2,112 +2,139 @@
 
 
-class CategoryUpdateMutation(
-        StaffMemberRequiredMixin, ModelFormUpdateMutation):
-    permissions = 'category.edit_category'
+class CategoryUpdateMutation(ModelMutation):
+    class Arguments:
+        id = graphene.ID(
+            required=True, description='ID of a category to update.')
+        input = CategoryInput(
+            required=True, description='Fields required to update a category.')
 
     class Meta:
-        description = 'Updates an existing category.'
-        form_class = CategoryForm
+        description = 'Updates a category.'
+        model = models.Category
 
     @classmethod
-    def get_form_kwargs(cls, root, info, **input):
-        kwargs = super().get_form_kwargs(root, info, **input)
-        kwargs['parent_pk'] = kwargs['instance'].parent_id
-        return kwargs
+    def user_is_allowed(cls, user, input):
+        return user.has_perm('category.edit_category')
 
 
-class CategoryDelete(StaffMemberRequiredMixin, ModelDeleteMutation):
-    permissions = 'category.edit_category'
+class CategoryDelete(ModelDeleteMutation):
+    class Arguments:
+        id = graphene.ID(
+            required=True, description='ID of a category to delete.')
 
     class Meta:
         description = 'Deletes a category.'
         model = models.Category
 
+    @classmethod
+    def user_is_allowed(cls, user, input):
+        return user.has_perm('category.edit_category')
+
+
+class CollectionInput(graphene.InputObjectType):
+    is_published = graphene.Boolean(
+        description='Informs whether a collection is published.')
+    name = graphene.String(description='Name of the collection.')
+    slug = graphene.String(description='Slug of the collection.')
+    products = graphene.List(
+        graphene.ID,
+        description='List of products to be added to the collection.')
+    background_image = Upload(description='Background image file.')
 
-class CollectionCreateMutation(StaffMemberRequiredMixin, ModelFormMutation):
-    permissions = 'collection.edit_collection'
+
+class CollectionCreateMutation(ModelMutation):
+    class Arguments:
+        input = CollectionInput(
+            required=True,
+            description='Fields required to create a collection.')
 
     class Meta:
         description = 'Creates a new collection.'
-        form_class = CollectionForm
+        model = models.Collection
 
     @classmethod
-    def get_form_kwargs(cls, root, info, **input):
-        product_ids = input.pop('products', None)
-        kwargs = super().get_form_kwargs(root, info, **input)
-        if product_ids:
-            products = set(get_nodes(product_ids, Product))
-            kwargs['data']['products'] = products
-        return kwargs
+    def user_is_allowed(cls, user, input):
+        return user.has_perm('collection.edit_collection')
 
 
-class CollectionUpdate(StaffMemberRequiredMixin, ModelFormUpdateMutation):
-    permissions = 'collection.edit_collection'
+class CollectionUpdate(ModelMutation):
```

Comment:
From what I see, right now there are two ways to add products to collection: using this mutation and `CollectionAddProducts` below.
Maybe one could modify `_clean_input` to remove proper key from dict; or, we decide that we update list of products every time, by dropping `CollectionAddProducts`.